File: JS, PHP, PYTHON/PYTHON/TABLA VERDAD.py
import itertools
# Definimos las variables
variables = ['A', 'B']
# Generamos todas las combinaciones de valores booleanos
combinaciones = list(itertools.product([True, False], repeat=len(variables)))
# Imprimimos la cabecera de la tabla
print(f"{' | '.join(variables)} | A AND B")
print("-" * 30)
# Evaluamos la expresión para cada combinación
for combinacion in combinaciones:
    A, B = combinacion
    resultado = A and B
    print(f"{A} | {B} | {resultado}")

File: JS, PHP, PYTHON/JAVASCRIPT/TABLA VERDAD.js
const variables = ['A', 'B'];

// Generamos todas las combinaciones de valores booleanos
const combinaciones = [];
for (let i = 0; i < Math.pow(2, variables.length); i++) {
    combinaciones.push([(i & 2) !== 0, (i & 1) !== 0]);
}

// Imprimimos la cabecera de la tabla
console.log(`${variables.join(' | ')} | A AND B`);
console.log('-'.repeat(30));

// Evaluamos la expresión para cada combinación
for (const combinacion of combinaciones) {
    const [A, B] = combinacion;
    const resultado = A && B;
    console.log(`${A} | ${B} | ${resultado}`);
}